superdesk/editor_utils.py

```diff
@@ -439,3 +615,70 @@ def prepend_block(self, block):
         else:
             index = 0
         self.blocks.insert(index, block)
+
+
+def _replace_text(content_state, old, new):
+    if not old:
+        return
+    for block in content_state['blocks']:
+        if block.get('type') == 'atomic':
+            entity = content_state[ENTITY_MAP][str(block[ENTITY_RANGES][0]['key'])]
+            if entity['type'] == 'TABLE':
+                cells = entity['data']['data']['cells']
+                for row in cells.values():
+                    for cell in row.values():
+                        _replace_text(cell, old, new)
+            continue
+        if not block.get('text'):
+            continue
+        start = 0
+        while True:
+            try:
+                index = block['text'].index(old, start)
+                start = index + len(old)
+                block['text'] = new.join([block['text'][:index], block['text'][start:]])
+                for range_field in (ENTITY_RANGES, INLINE_STYLE_RANGES):
+                    if block.get(range_field):
+                        ranges = []
```

Comment:
we might lose all style ranges at this point.
Example:
```
item = {
    'body_html': '<h1>head</h1>\n<p>lorem <b>this is bold</b> and <b>bold</b> end</p>',
    'fields_meta': {}
}
replace_text(item, 'body_html', 'bold', 'UL')
pprint(item)
```

Reply:
I've fixed it to keep styles